File: LeetCode/705. Design HashSet/705.c
#include <stdbool.h>
#include <stdlib.h>

typedef struct {
    bool* set;
} MyHashSet;


MyHashSet* myHashSetCreate() {
    MyHashSet* obj = (MyHashSet*)malloc(sizeof(MyHashSet));
    obj->set = (bool*)calloc(1000001, sizeof(bool)); // Initialize a boolean array of size 1000001
    return obj;
}

void myHashSetAdd(MyHashSet* obj, int key) {
    obj->set[key] = true;
}

void myHashSetRemove(MyHashSet* obj, int key) {
    obj->set[key] = false;
}

bool myHashSetContains(MyHashSet* obj, int key) {
    return obj->set[key];
}

void myHashSetFree(MyHashSet* obj) {
    free(obj->set);
    free(obj);
}

/**
 * Your MyHashSet struct will be instantiated and called as such:
 * MyHashSet* obj = myHashSetCreate();
 * myHashSetAdd(obj, key);
 * myHashSetRemove(obj, key);
 * bool param_3 = myHashSetContains(obj, key);
 * myHashSetFree(obj);
*/

File: LeetCode/705. Design HashSet/705.cs
public class MyHashSet {

    private bool[] set;

    /** Initialize your data structure here. */
    public MyHashSet() {
        set = new bool[1000001]; // Initialize a boolean array of size 1000001
    }
    
    /** Add a key to the set. */
    public void Add(int key) {
        set[key] = true;
    }
    
    /** Remove a key from the set. */
    public void Remove(int key) {
        set[key] = false;
    }
    
    /** Returns true if the set contains the specified key. */
    public bool Contains(int key) {
        return set[key];
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.Add(key);
 * obj.Remove(key);
 * bool param_3 = obj.Contains(key);
 */
